src/UserSpaceInstrumentation/DummyProcessForTest.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include "DummyProcessForTest.h"
+
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "absl/time/clock.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace fs = std::filesystem;
+
+namespace {
+
+// Create a file at path.
+void Touch(fs::path p) {
+  std::ofstream ofs(p);
+  ofs << "\n";
+  ofs.close();
+}
+
+}  // namespace
+
+DummyProcessForTest::DummyProcessForTest() {
+  flag_file_run_child_ = std::tmpnam(nullptr);
+  flag_file_child_started_ = std::tmpnam(nullptr);
+  Touch(flag_file_run_child_);
+  pid_ = fork();
+  CHECK(pid_ != -1);
+  // Start the workload and have the parent wait for the startup to complete.
+  if (pid_ == 0) {
+    DummyWorkload();
+    exit(0);
+  } else {
+    while (!fs::exists(flag_file_child_started_)) continue;
+  }
+}
+
+DummyProcessForTest::~DummyProcessForTest() {
+  fs::remove(flag_file_child_started_);
```

Comment:
Can it happen that after this is executed, `Touch(flag_file_child_started_);` in `DummyWorkload()` is called again (basically a race condition)? Or does this rely on the fact that the forked process will also run the destructor?

Reply:
Good catch! From my computer:
```
> ls /tmp/file* -la | wc -l
5270
```
The `fs::remove(flag_file_child_started_);` needs to go after the waitpid.

fwiw: The child is not executing the destructor. It calls `exit(0)` after the workload is done and that only triggers static destruction.